pkg/api/types_test.go

```diff
@@ -416,3 +417,82 @@ loop:
 		t.Fatalf("mismatch between internal and admin API fields: afield=%q, ifield=%q", afield, ifield)
 	}
 }
+
+func Test20180930previewAPIParity(t *testing.T) {
+	// the algorithm is: list the fields of both types.  If the head of either
```

Comment:
Good comment!

Reply:
All credit to @jim-minter who wrote the Admin API parity test which I copied and adapted.